ibm/data_source_ibm_cm_offering.go

```diff
@@ -0,0 +1,283 @@
+				Computed:    true,
+				Description: "URL to be displayed in the Consumption UI for getting support on this offering.",
+			},
+			"tags": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "List of tags associated with this catalog.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"short_description": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Short description in the requested language.",
+			},
+			"long_description": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Long description in the requested language.",
+			},
+			"permit_request_ibm_public_publish": &schema.Schema{
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Is it permitted to request publishing to IBM or Public.",
+			},
+			"ibm_publish_approved": &schema.Schema{
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Indicates if this offering has been approved for use by all IBMers.",
+			},
+			"public_publish_approved": &schema.Schema{
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Indicates if this offering has been approved for use by all IBM Cloud users.",
+			},
+			"public_original_crn": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The original offering CRN that this publish entry came from.",
+			},
+			"publish_public_crn": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn of the public catalog entry of this offering.",
+			},
+			"portal_approval_record": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The portal's approval record ID.",
+			},
+			"portal_ui_url": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The portal UI URL.",
+			},
+			"catalog_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The id of the catalog containing this offering.",
+			},
+			"catalog_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the catalog.",
+			},
+			"disclaimer": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "A disclaimer for this offering.",
+			},
+			"hidden": &schema.Schema{
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Determine if this offering should be displayed in the Consumption UI.",
+			},
+			"repo_info": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Repository info for offerings.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"token": &schema.Schema{
+							Type:        schema.TypeString,
+							Optional:    true,
```

Comment:
when parent attribute is computed.. child attributes cannot be optional.. it should computed..